services/QuillLMS/app/services/units/assignment_helpers.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Units::AssignmentHelpers
```

Comment:
All of this code is from the `AssignRecommendationsWorker`, I just pulled it out because we want the same logic for the post-test assignment.

Reply:
Long term, I think this should be proper service object[s], but this is definitely an improvement on extracting out some duplicated logic.